From 8c6e4b688cc7feade17caa832f9304262025fd17 Mon Sep 17 00:00:00 2001
From: Qian Lei <qianl.fnst@cn.fujitsu.com>
Date: Tue, 27 Jan 2015 19:34:26 +0800
Subject: [PATCH] Dont compile extensions

Signed-off-by: Qian Lei <qianl.fnst@cn.fujitsu.com>
---
 json.gemspec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/json.gemspec b/json.gemspec
index 75926b1..af7e9ca 100644
--- a/json.gemspec
+++ b/json.gemspec
@@ -13 +13 @@ Gem::Specification.new do |s|
-  s.extensions = ["ext/json/ext/generator/extconf.rb", "ext/json/ext/parser/extconf.rb"]
+#  s.extensions = ["ext/json/ext/generator/extconf.rb", "ext/json/ext/parser/extconf.rb"]
-- 
1.8.3.1

