diff --git a/core/hash.c b/core/hash.c
index 5940f66..bf989fc 100644
--- a/core/hash.c
+++ b/core/hash.c
@@ -42,13 +42,16 @@ static uint32_t murmur2_hash(char *key, uint64_t keylen) {
 	switch (keylen) {
 		case 3:
         		h ^= key[2] << 16;
+			break;
 			/* fallthrough */
     		case 2:
         		h ^= key[1] << 8;
+			break;
 			/* fallthrough */
     		case 1:
         		h ^= key[0];
         		h *= 0x5bd1e995;
+			break;
     	}
 
 	h ^= h >> 13;
diff --git a/core/routing.c b/core/routing.c
index 4ba70b4..0b05643 100644
--- a/core/routing.c
+++ b/core/routing.c
@@ -1792,10 +1792,10 @@ static int uwsgi_route_condition_ipv6in(struct wsgi_request *wsgi_req, struct uw
 
 	int i = (pfxlen / 32);
 	switch (i) {
-	case 0: mask[0] = 0; /* fallthrough */
-	case 1: mask[1] = 0; /* fallthrough */
-	case 2: mask[2] = 0; /* fallthrough */
-	case 3: mask[3] = 0; /* fallthrough */
+	case 0: mask[0] = 0; break; /* fallthrough */
+	case 1: mask[1] = 0; break; /* fallthrough */
+	case 2: mask[2] = 0; break; /* fallthrough */
+	case 3: mask[3] = 0; break; /* fallthrough */
 	}
 
 	if (pfxlen % 32)
