Index: stevedore-0.8/setup.py
===================================================================
--- stevedore-0.8.orig/setup.py
+++ stevedore-0.8/setup.py
@@ -1,8 +1,8 @@
 #!/usr/bin/env python
 
 # Bootstrap installation of Distribute
-import distribute_setup
-distribute_setup.use_setuptools()
+#import distribute_setup
+#distribute_setup.use_setuptools()
 
 from setuptools import setup, find_packages
 
@@ -11,8 +11,7 @@ try:
 except IOError:
     long_description = ''
 
-install_requires = ['distribute',
-                    ]
+install_requires = []
 
 setup(
     name='stevedore',
